runelite-client/src/main/java/net/runelite/client/ui/overlay/worldmap/WorldMapOverlayMouseListener.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2018, Morgan Lewis <https://github.com/MESLewis>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.ui.overlay.worldmap;
+
+import java.awt.Rectangle;
+import java.awt.event.MouseEvent;
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+import net.runelite.api.Client;
+import net.runelite.api.Point;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.input.MouseListener;
+import net.runelite.client.input.MouseManager;
+
+@Singleton
+public class WorldMapOverlayMouseListener extends MouseListener
+{
+	private final Provider<Client> clientProvider;
+	private final MouseManager mouseManager;
+	private final WorldMapPointManager worldMapPointManager;
+
+	@Inject
+	public WorldMapOverlayMouseListener(Provider<Client> clientProvider, MouseManager mouseManager, WorldMapPointManager worldMapPointManager)
+	{
+		this.clientProvider = clientProvider;
+		this.mouseManager = mouseManager;
+		this.worldMapPointManager = worldMapPointManager;
+		mouseManager.registerMouseListener(0, this);
+	}
+
+	@Override
+	public MouseEvent mousePressed(MouseEvent e)
+	{
+		if (e.getButton() == MouseEvent.BUTTON1)
```

Comment:
Use SwingUtillities.isLeftMouseButton (or something like that, not 100% sure about the method name now)